packages/purpose-process/src/services/purposeService.ts

```diff
@@ -1186,6 +1185,8 @@ export function purposeServiceBuilder(
 
       const purposeToClone = await retrievePurpose(purposeId, readModelService);
 
+      assertRequesterCanActAsConsumer(purposeToClone.data, authData, undefined);
+
```

Comment:
We should also make sure, if the purpose is delegated, to clone also the delegationId (below 👇)